hw/ip/pattgen/doc/_index.md

```diff
@@ -0,0 +1,137 @@
+---
+title: "PATTGEN HWIP Technical Specification"
+---
+
+# Overview
+
+This document specifies PATTGEN hardware IP (HWIP) functionality.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification/index.md" >}})
+See that document for integration overview within the broader top-level system.
+
+## Features
+
+- Generates time-dependent patterns on two (2) channels, each with its own clock.
+   - In each channel, data is transmitted serially on a one-bit data (`pda`) output, which is synchronous to a corresponding parallel clock signal (`pcl`).
+   - The channels may operate independently or synchronously with each other.
+- Each output channel supports the following configuration registers:
+    - Pattern data per output (up to 64 bits of data).
+    - Pattern length per output (between 1 and 64 bit).
+    - 32-bit pre-divider to derive pattern clock from I/O clock (minimum ratio: 2).
+    - A 10-bit repeat counter
+    - A bit to control the polarity of the clock signal
+- Sends interrupt on pattern completion.
+## Description
+
+The PATTGEN IP transmits short (maximum 64 bits) time-dependent data patterns on two clock-parallel channels.
+Each channel consists of one clock (`pcl`) and one data (`pda`) line.
+The output channels may be activated and operated independently, or they can be started at the same time to effectively create a 4-output pattern.
+
+## Compatibility
+
+This IP block does not have any direct hardware compatibility requirements.
+
+# Theory of Operations
+
+The pattern can be started (or halted) on either channel by setting the corresponding `CTRL.ENABLE` bit to 1 (or 0) for the desired channel.
+Once disabled, either channel can be configured independently so long as the desired channel is first disabled.
+The channel parameters (i.e. clock divider ratio, clock polarity, pattern length, pattern data, and repetition count) can all be programmed on a per-channel basis.
+Enabling the pattern generator channel starts the pattern from the beginning.
+
+Please note that writes to a channel's configuration registers have no effect while the channel is enabled.
+For operational simplicity, the configuration registers are only transferred into the internal finite state machines while a channel is disabled.
+Changes to the configuration registers only take effect once the channel has been disabled and re-enabled.
+
+## Block Diagram
+
+![](pattgen_block_diagram.svg)
```

Comment:
In this diagram: please double-check the output names with the modified register and interrupt names. At least the done interrupts are now named differently.

Reply:
Ugh, well I did make a scan of the registers.  However I think you are right, I may have neglected the interrupts.  Thanks.